jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileReader.kt

```diff
@@ -0,0 +1,99 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.credentials.profiles
+
+import com.intellij.util.text.nullize
+import software.amazon.awssdk.profiles.Profile
+import software.amazon.awssdk.profiles.ProfileFile
+import software.amazon.awssdk.profiles.ProfileProperty
+import software.aws.toolkits.resources.message
+
+data class Profiles(val validProfiles: Map<String, Profile>, val invalidProfiles: Map<String, Exception>)
+
+/**
+ * Reads the AWS shared credentials files and produces what profiles are valid and if not why it is not
+ */
+fun validateAndGetProfiles(): Profiles {
+    val allProfiles: Map<String, Profile> = ProfileFile.defaultProfileFile().profiles()
+
+    val validProfiles = mutableMapOf<String, Profile>()
+    val invalidProfiles = mutableMapOf<String, Exception>()
+
+    allProfiles.values.forEach {
+        try {
+            validateProfile(it, allProfiles)
+            validProfiles[it.name()] = it
+        } catch (e: Exception) {
+            invalidProfiles[it.name()] = e
+        }
+    }
+
+    return Profiles(validProfiles, invalidProfiles)
+}
+
+private fun validateProfile(profile: Profile, allProfiles: Map<String, Profile>) {
+    when {
+        profile.propertyExists(ProfileProperty.ROLE_ARN) -> validateAssumeRoleProfile(profile, allProfiles)
```

Comment:
question: does this normalize case somewhere?

Reply:
Nope, the Java SDK doesn't normalize

Reply:
Oh, true